scipy/sparse/_construct.py

```diff
@@ -66,10 +79,17 @@ def spdiags(data, diags, m=None, n=None, format=None):
         m = n = len(data[0])
     elif n is None:
         m, n = m
-    return dia_matrix((data, diags), shape=(m, n)).asformat(format)
+    D = dia_matrix((data, diags), shape=(m, n)).asformat(format)
+    if sparray is None:
```

Comment:
Is the idea to make this default explicit here? I think I'd prefer to leave this out for now, and when the time comes to change the default we can insert new code to do so.

Reply:
Do you mean that you would prefer
```
 def spdiags(data, diags, m=None, n=None, format=None, sparray=False): 
```

Reply:
and
```
D = dia_matrix((data, diags), shape=(m, n)).asformat(format)
if sparray:
    return _fmt_to_sparray[D.format](D)
return D
```